### PR TITLE
chore: improve contributor experience

### PR DESCRIPTION
**Pull request by nklayman**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

Updates the setup scripts to install packages with the package manager instead of through aliases. Updates the api example scripts to allow for tauri build and dev. Revamps the contributor guide. Replaces the communication example with a simple helloworld app, as the API example contains all of the communication app functionality.

### CODE REVIEW

File: .scripts/setup.ps1 [hunk 1]
```diff
@@ -4,27 +4,22 @@ cd api
...
+$result = $host.ui.PromptForChoice("Node.js CLI", "Do you want to install the Node.js CLI?", $options, 1)
 switch ($result) {
   0{
     cd cli\tauri.js
-    yarn; yarn build
+    yarn; yarn build; yarn link
     cd ..\..
-    Set-Alias stauri "$(pwd)\cli\tauri.js\bin\tauri.js"
-    echo "Added alias 'stauri' for '$(pwd)\cli\tauri.js\bin\tauri.js'"
     echo "Tauri Node.js CLI installed. Run it with '$ stauri [COMMAND]'"
```
Comment (amrbashir) [inline .scripts/setup.ps1]:
```suggestion
    echo "Tauri Node.js CLI installed. Run it with '$ tauri [COMMAND]'"
```

Comment (nklayman) [reply .scripts/setup.ps1]: Good catch!

**Review by lucasfernog — APPROVED** [review]

**Comment by lucasfernog** [comment]
@nklayman is this PR ready to be merged? looks good to me :)

**Comment by Madsadglad** [comment]
Huh

**Pull request merged**